tool/teleport/common/debug.go

```diff
@@ -0,0 +1,252 @@
+	path := constants.PProfEndpointsPrefix + profileName
+	if seconds > 0 {
+		path += fmt.Sprintf("?seconds=%d", seconds)
+	}
+
+	resp, err := c.do(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	result, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, trace.BadParameter("Unable to collect profile %q: %s", profileName, result)
+	}
+
+	return result, nil
+}
+
+func (c *debugServiceClient) do(ctx context.Context, method, path string, body []byte) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, method, "http://debug"+path, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	resp, err := c.clt.Do(req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return resp, nil
+}
+
+func onSetLogLevel(configPath string, level string) error {
+	ctx := context.Background()
+
+	if contains := slices.Contains(logutils.SupportedLevelsString, strings.ToUpper(level)); !contains {
+		return trace.BadParameter("%q log level not supported", level)
+	}
+
+	clt, err := newDebugServiceClient(configPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	setMessage, err := clt.SetLogLevel(ctx, level)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Println(setMessage)
+	return nil
+}
+
+func onGetLogLevel(configPath string) error {
+	ctx := context.Background()
+	clt, err := newDebugServiceClient(configPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	level, err := clt.GetLogLevel(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Printf("Current log level %q\n", level)
+	return nil
+}
+
+// supportedProfiles list of supported pprof profiles that can be collected.
+// This list is composed by runtime/pprof.Profile and http/pprof definitions.
+var supportedProfiles = map[string]struct{}{
+	"allocs":       {},
+	"block":        {},
+	"cmdline":      {},
+	"goroutine":    {},
+	"heap":         {},
+	"mutex":        {},
+	"profile":      {},
+	"threadcreate": {},
+	"trace":        {},
+}
+
+// defaultCollectProfiles defines the default profiles to be collected in case
+// none is provided.
+var defaultCollectProfiles = []string{"goroutine", "heap", "profile"}
+
+func onCollectProfile(configPath string, rawProfiles string, seconds int, out io.Writer) error {
+	ctx := context.Background()
+
+	profiles := defaultCollectProfiles
+	if rawProfiles != "" {
+		profiles = strings.Split(rawProfiles, ",")
+	}
+
+	for _, profile := range profiles {
+		if _, ok := supportedProfiles[profile]; !ok {
+			return trace.BadParameter("%q profile not supported", profile)
+		}
+	}
+
+	clt, err := newDebugServiceClient(configPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var output bytes.Buffer
+	if err := createProfilesArchive(ctx, clt, &output, profiles, seconds); err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Fprintln(out, output.String())
```

Comment:
```suggestion
	fmt.Fprint(out, output.String())
```

For me, untarring the file gives the following error (not sure if it's because my tar is old?):
```bash
$ tar -xzf debug.tar.gz

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now

$ tar --version
tar (GNU tar) 1.34
Copyright (C) 2021 Free Software Foundation, Inc.
```
I think it might be related to the extra line end we added here.

Reply:
I'll drop the line. But I couldn't reproduce it directly. My GNU tar gives some warning about trailing garbage, but it extracts.

Reply:
yes it extracts fine for me too, after spitting the above error. the error is confusing though how far it goes.